validate.go

```diff
@@ -71,3 +71,34 @@ func ComposeValidators(validators ...Validator) Validator {
 		return nil
 	}
 }
+
+// isZero returns true if the passed value is the zero object
+func isZero(v reflect.Value) bool {
```

Comment:
Not sure this implantation covers all the caveats that make a zero object.
This is what I have used in the past:
```
reflect.DeepEqual(v.Interface(), reflect.Zero(v.Type()).Interface())
```
Basically we just create a zero value of the type passed in and verify that equals the actual value passed in.  Not saying your code wont work, clearly it does, but I am just not sure what (if any) edge cases are missing.  I am pretty confident the DeepEqual covers all the edge cases the the cost of an extra zero-allocation.  If you want to keep your implementation that is fine, just something to think about.